libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -112,21 +192,31 @@ def create_audio_file(
     filename: str,
     assets_directory: StrPath,
     overwrite: bool = True,
+    # TODO: Once assets and cached-assets are migrated to S3, this parameter is no more needed
+    use_s3_storage: bool = False,
+    s3_client: Optional[S3Client] = None,
```

Comment:
instead of number of flat arguments, maybe we could have `storage_options: DirectoryStorageOptions | S3StorageOptions`, and pack all the S3 related options inside `S3StorageOptions`.

Also if we migrate to S3 in production, we can let the support for directory-based storage too?